nbgitpuller/templates/status.html

```diff
@@ -7,11 +7,12 @@
 data-path="{{ path | urlencode }}"
 {% if branch %}data-branch="{{ branch | urlencode }}"{% endif %}
 {% if depth %}data-depth="{{ depth | urlencode }}"{% endif %}
+{% if provider %}data-provider="{{ provider | urlencode }}"{% endif %}
```

Comment:
contentProvider I guess.

Reply:
> * I'm not that experienced with Python and I've never used the `global` keyword, but I recall it is a practice to >avoid. I think that is plausible to avoid, but let's not bother unless someone with more experience can point us >. >towards a concrete alternative solution.

I went all over the place on this and ended with the global variable solution from this [reference](https://stackoverflow.com/questions/64948770/how-to-save-a-new-value-to-a-variable-outside-an-async-function-inside-that-asy).  I would not call it definitive but got me through!

> * I'm a bit concerned about seeing that we seem to rely on a single temp downloader directory. Can we instead rely on something like the following ideas to avoid conflicts?
>   
>   1. A root location for temp downloads, but we have separate folders under it to avoid conflicts
>   2. Use of a Python [tempfile.TemporaryDirecory](https://docs.python.org/3/library/tempfile.html#tempfile.TemporaryDirectory)
> Btw, if we use a TemporaryDirectory, I think you can pass `dir=...` and make it create the temporary directory within some other directory.

I got this in -- good deal